loader/linux/icd_linux.c

```diff
@@ -143,6 +143,16 @@ void khrIcdOsVendorsEnumerate(void)
         }
 
         closedir(dir);
+
+        KHRicdVendor *vendorIterator;
+        for (vendorIterator = khrIcdVendors; vendorIterator; vendorIterator = vendorIterator->next)
+        {
+            if (vendorIterator->libName != NULL)
+            {
+                free(vendorIterator->libName);
+                vendorIterator->libName = NULL;
+            }
+        }
```

Comment:
Code is duplicated here and in `icd_windows.c`. Also this is cleaning up something created in `icd.c` and is platform independent. This should go in it's own function defined in `icd.c` and called from here.